KeenClient/KIOEventStore.m

```diff
@@ -685,6 +689,12 @@ - (void)closeDB {
 }
 
 - (id)convertNSDateToISO8601:(id)date {
+    __block NSString *iso8601 = nil;
```

Comment:
@terryhorner I think we can just reopen it after a failure occurs. At least that sounds like the optimal way to do it. 

I'm gonna do some research if that's okay, or if it's really best for us to open/close on every transaction.